caffe2/python/muji.py

```diff
@@ -39,11 +46,14 @@ def Allreduce(net, blobs, reduced_affix="_reduced", gpu_indices=None):
             "gpu_indices length and blobs length mismatch: %d vs %d" %
             (len(gpu_indices), len(blobs))
         )
-    if len(blobs) == 2:
+    pattern = workspace.GetCudaPeerAccessPattern()
```

Comment:
Any chance this method could be called on CPU? Seems like the answer is no, but confirming. Also, we will eventually be supporting HIP/AMD, so I wonder if this is safe. cc @bddppq for that.

Reply:
The plan for AMD multi-gpu support is still unsettled, and very likely we will need to special case/fork the code anyway, so I would say let's don't consider that for now.